lib/phoenix_live_view/plug.ex

```diff
@@ -0,0 +1,59 @@
+defmodule Phoenix.LiveView.Plug do
+  @moduledoc """
+   Provides a Plug for LiveView rendering.
+
+   ## Options
+
+   See `Phoenix.LiveView.Router.live/3` for full set of supported options.
+
+   ## Examples
+
+       plug Phoenix.LiveView.Plug, MyApp.UserLive
+       plug Phoenix.LiveView.Plug, {MyApp.UserLive, session: [:path_params, :user_id]}
+
+  """
+
+  alias Phoenix.LiveView.Controller
+  alias Plug.Conn
+
+  @behaviour Plug
+
+  @impl Plug
+  def init(opts), do: opts
+
+  @impl Plug
+  @spec call(Conn.t(), module()) :: Conn.t()
+  def call(%Conn{} = conn, view) when is_atom(view) do
+    opts = Map.get(conn.private, :phoenix_live_view, [])
+    do_call(conn, view, opts)
+  end
+
+  @spec call(Conn.t(), {module(), Keyword.t()}) :: Conn.t()
+  def call(%Conn{} = conn, {view, opts}) when is_atom(view) and is_list(opts) do
+    do_call(conn, view, opts)
+  end
+
+  defp do_call(conn, view, opts) do
+    session_opts = Keyword.get(opts, :session, [:path_params])
+    render_opts = Keyword.merge(opts, session: session(conn, session_opts))
+
+    conn
+    |> put_new_layout_from_router()
+    |> Controller.live_render(view, render_opts)
+  end
+
+  defp session(conn, session_opts) do
+    Enum.reduce(session_opts, %{}, fn
+      :path_params, acc -> Map.put(acc, :path_params, conn.path_params)
+      key, acc -> Map.put(acc, key, Conn.get_session(conn, key))
+    end)
+  end
+
+  defp put_new_layout_from_router(conn) do
+    if layout_view = conn.private[:phoenix_live_view_default_layout] do
```

Comment:
What if we put `phoenix_live_view_default_layout` as a `:default_layout` option inside `conn.private[:phoenix_live_view]`? this way we always store only a single key in private, instead of multiple.